spec/money_spec.rb

```diff
@@ -80,6 +80,26 @@
     expect(non_fractional_money.to_s(:amount)).to eq("1")
   end
 
+  it "to_s rounds when  more fractions than currency allows" do
+    expect(Money.new("9.999", "USD").to_s).to eq("10.00")
+    expect(Money.new("9.889", "USD").to_s).to eq("9.89")
+  end
+
+  it "to_s does not round when fractions same as currency allows" do
+    expect(Money.new("1.25", "USD").to_s).to eq("1.25")
+    expect(Money.new("9.99", "USD").to_s).to eq("9.99")
+    expect(Money.new("9.999", "BHD").to_s).to eq("9.999")
+  end
+
+  it "to_s does not round if amount is larger than float allows" do
+    expect(Money.new("99999999999999.99", "USD").to_s).to eq("99999999999999.99")
+    expect(Money.new("999999999999999999.99", "USD").to_s).to eq("999999999999999999.99")
```

Comment:
please also add `99999999999999.99` decimal use case (will require you to change the precision from 15 to 16)

Reply:
hmm, you mean 

```ruby
expect(Money.new(BigDecimal( 99999999999999.99, 16), "USD").to_s).to eq("999999999999999999.99")
```

or change `Money::Helpers#value_to_decimal` to use `BigDecimal(num, 16)` instead of `BigDecimal(num, Float::DIG)`?

Reply:
just adding, to catch the case where this was changed to 10000...
```ruby
expect(Money.new(99999999999999.99, "USD").to_s).to eq("99999999999999.99")
```